src/app/modules/item/components/item-header/access-code-view/access-code-view.component.ts

```diff
@@ -0,0 +1,83 @@
+import { Component } from '@angular/core';
+import { ConfirmationService, MessageService } from 'primeng/api';
+import { ERROR_MESSAGE } from 'src/app/shared/constants/api';
+import { TOAST_LENGTH } from 'src/app/shared/constants/global';
+import { InvalidCodeReason, JoinByCodeService } from '../../../http-services/join-by-code.service';
+
+@Component({
+  selector: 'alg-access-code-view',
+  templateUrl: './access-code-view.component.html',
+  styleUrls: [ './access-code-view.component.scss' ]
+})
+export class AccessCodeViewComponent {
+
+  state: 'ready'|'loading' = 'ready';
+
+  constructor(
+    private joinByCodeService: JoinByCodeService,
+    private messageService: MessageService,
+    private confirmationService: ConfirmationService,
+  ) { }
+
+  onClickAccess(groupCode: string): void {
+    this.state = 'loading';
+
+    this.joinByCodeService.checkCodeValidity(groupCode).subscribe(response => {
+      this.state = 'ready';
+      if (!response.valid) {
+        this.errorToast(this.invalidCodeReasonToString(response.reason));
+        return;
+      }
+      this.confirmationService.confirm({
+        header: $localize`Join the group "${response.group.name}"`,
+        message: $localize`Are you sure you want to join the group "${response.group.name}"?`,
+        acceptLabel: $localize`Join`,
+        acceptIcon: 'fa fa-check',
+        rejectLabel: $localize`Cancel`,
+        accept: () => {
+          this.joinGroup(groupCode);
+        }
+      });
+    });
+  }
+
+  joinGroup(code: string): void {
+    this.joinByCodeService.joinGroupThroughCode(code).subscribe(
+      _result => this.successToast(),
+      _err => this.errorToast()
+    );
+  }
+
+  invalidCodeReasonToString(reason: InvalidCodeReason): string {
+    switch (reason) {
+      case 'already_member':
+        return $localize`You are already a member of this group`;
+      case 'conflicting_team_participation':
+        return $localize`There is conflicting group participation`;
+      case 'frozen_membership':
+        return $localize`Your membership is frozen`;
```

Comment:
I am not sure of the meaning of `conflicting_team_participation` and `frozen_membership` so I don't know if the message is really accurate, same goes for the french translation of these two

Reply:
You can use:
- conflicting_team_participation :  "You cannot join this team as it would conflict with another team you belong to."
- frozen_membership: "This group does not allow any membership change."
- team_conditions_not_met: "You cannot join this team as it would break the entry conditions of content it is participating to."

Reply:
fixed